### DISCUSSION TITLE
Web Component to Generate Image Color Palettes

### DISCUSSION THREAD

**Discussion by giscus** [cfjedimaster/raymondcamden2023#29] [Comments]
# Web Component to Generate Image Color Palettes

A web component that adds a palette to images.

https://www.raymondcamden.com/2024/07/16/web-component-to-generate-image-color-palettes

**Comment by JonasEriksson** [comment]
Love stuff like this - thanks for sharing! As you indicated, no urgent immediate *need* comes to mind, but cool execution and would totally see this as a cool plugin for something like Masa CMS in the CFML world.

**Reply by cfjedimaster** [reply]
One real world use for ColorThief, _not_ this web component, would be perhaps a modal view of the image where the background complements the primary colors.  I should try to build that.

**Comment by briankoser** [comment]
Fun web component!

The whitespace after the img happens because img is inline, so it includes space for descenders (the dangly parts of g, j, p, q). 

`display:block` removes the whitespace.

https://codepen.io/briankoser/pen/BagjLpY

**Reply by cfjedimaster** [reply]
Wow, so simple, thank you!
